### DISCUSSION TITLE
Is Cookie-AutoDelete still necessary with Firefox Total Cookie Protection?

### DISCUSSION THREAD

**Discussion by twocoolbeans** [Cookie-AutoDelete/Cookie-AutoDelete#1056] [General] [answered]
Firefox 86.0 just released Total Cookie Protection, basically putting cookies in separate containers for each domain. Firefox also blocks supercookies and uses a container system for the cache. With this in mind is Cookie-AutoDelete still necessary from a privacy perspective and why?

Have loved and used this extension for years, but also am a minimalist who tries to keep things simple. Fewer extensions to configure keeps things simple.

**Comment by kennethtran93** [comment, marked as answer]
Ultimately it is up to the user.  

While Firefox may have added additional protections, CAD would still be cleaning cookies, especially third-party and unwanted cookies regardless.  If you truly want privacy, then you'd probably want most cookies removed as soon as possible from sites you visit except for those you configured to be kept, regardless of the browser's cookie protection, so that if you do visit that site again in the same session, chances are minimal that they will be able to connect your latest visit with your previous one (provided the cookies were cleaned up after the site is non-existent in your active tabs).

If I summarize it correctly, the cookie protection, depending on what the browser considers third-party tracking, may still allow those to track how many times you have visited that specific site, just not being able to link it to anything else.  Even the site itself may keep track of the times you visited using cookies.  Would you rather "User A visited this site N times" or "User X/Y/Z visited this site 1 time", when you visit that site multiple times?

**Reply by twocoolbeans** [reply]
@kennethtran93 Good points. While, I was aware of the example you stated, you convinced me that configuring a few more settings is worth preventing "User A visited this site N times."

What are your thoughts about clearing the cache and other web data along with cookies? Do you see any reason why this is necessary from a privacy perspective if supercookies are not a concern?

**Reply by kennethtran93** [reply]
Sites may be able to store some form of unique identification on cache, localstorage, indexedDB and other web data types as well, so if it is not a site you want data to be kept for, just clean them away.

**Reply by twocoolbeans** [reply]
The same situation ("User A visited this site N times" or "User X/Y/Z visited this site 1 time") applies to cache, localstorage, indexedDB and other web data types (collectively referred to above as "supercookies") as well as regular cookies.

In light of the new Firefox features, decided to keep Cookie-AutoDelete. Firefox now seems to effectively block cross site tracking by default, but the site itself can still track users across visits. It is this latter scenario that I think Cookie-AutoDelete is still useful for.

**Reply by Bout2GitIt** [reply]
One thing that still confuses me here:
The CAD Settings menu suggests that using the Automatic Cleaning function with Container Tabs is undesirable. 
("WARNING: Enabling Container Tabs with any of the additional browsing data cleanups above may cause unwanted side effects" Automatic Cleaning is listed above this message.)

Am I reading this correctly? Do we have to choose between using CAD's Automatic Delete function and Firefox's Container Tabs?

**Reply by kennethtran93** [reply]
@Bout2GitIt Please start a new discussion.  What you are asking is not as relevant to the original question.
